src/components/Palette/Palette.tsx

```diff
@@ -0,0 +1,287 @@
+            event.stopPropagation();
+        },
+        [onBlur],
+    );
+
+    return (
+        <div
+            {...containerProps}
+            role={'grid'}
+            ref={handleRef}
+            className={paletteClassNames.palette({size}, className)}
+            onFocus={focusedOptionIndex === undefined ? onFocusGrid : undefined}
+            onBlur={onBlurGrid}
+            style={style}
+            data-qa={qa}
+            tabIndex={focusedOptionIndex === undefined ? 0 : -1}
+        >
+            {rows.map((row, rowNumber) => (
+                <div
+                    className={paletteClassNames.row(rowClassName)}
+                    key={`row-${rowNumber}`}
+                    role={'row'}
+                >
+                    {row.map((option) => (
+                        <PaletteOptionItem
+                            key={option.value}
+                            value={currentValue}
+                            option={option}
+                            checked={currentValue?.includes(option.value) ?? false}
+                            disabled={disabled}
+                            size={size}
+                            optionClassName={optionClassName}
+                            iconClassName={iconClassName}
+                            onUpdate={setCurrentValue}
+                            onFocus={onFocusOption}
+                            onBlur={onBlurOption}
+                        />
+                    ))}
+                </div>
+            ))}
+        </div>
+    );
+}) as PaletteComponent;
+
+Palette.displayName = 'Palette';
+
+function getRows(options: PaletteOption[], columns: number): PaletteOption[][] {
+    const rows: PaletteOption[][] = [];
+    let row: PaletteOption[] = [];
+
+    let column = 0;
+    for (const option of options) {
+        row.push(option);
+        column += 1;
+        if (column === columns) {
+            rows.push(row);
+            row = [];
+            column = 0;
+        }
+    }
+
+    if (row.length > 0) {
+        rows.push(row);
+    }
+
+    return rows;
+}
+
+function PaletteOptionItem({
+    value,
+    option,
+    checked,
+    disabled,
+    size,
+    optionClassName,
+    iconClassName,
+    onUpdate,
+    onFocus,
+    onBlur,
+}: {option: PaletteOption; checked: boolean} & Pick<
+    PaletteProps,
+    | 'value'
+    | 'disabled'
+    | 'size'
+    | 'optionClassName'
+    | 'iconClassName'
+    | 'onUpdate'
+    | 'onFocus'
+    | 'onBlur'
+>) {
+    const onUpdateOption = React.useCallback(() => {
+        if (!onUpdate) return;
+
+        if (value) {
+            const newValue = value.includes(option.value)
+                ? value.filter((v) => v !== option.value)
+                : [...value, option.value];
+
+            onUpdate(newValue);
+        } else {
+            onUpdate([option.value]);
+        }
+    }, [onUpdate, option.value, value]);
+
+    return (
+        <PaletteControl
```

Comment:
I think we do not need `PaletteControl` at all. Looks like redundant abstraction

Reply:
You're right
It's much simpler without it
Removed it